lib/events/athena/querier.go

```diff
@@ -0,0 +1,451 @@
+	qb.Append(tablename)
+	qb.Append(` WHERE event_date BETWEEN date(?) AND date(?)`, withTicks(params.fromUTC.Format(iso8601DateFormat)), withTicks(params.toUTC.Format(iso8601DateFormat)))
+	qb.Append(` AND event_time BETWEEN ? and ?`,
+		fmt.Sprintf("timestamp '%s'", params.fromUTC.Format(athenaTimestampFormat)), fmt.Sprintf("timestamp '%s'", params.toUTC.Format(athenaTimestampFormat)))
+
+	if params.sessionID != "" {
+		qb.Append(" AND session_id = ?", withTicks(params.sessionID))
+	}
+
+	if len(params.filter.eventTypes) > 0 {
+		// Athena does not support IN with single `?` and multiple parameters.
+		// Based on number of eventTypes, first query is prepared with defined
+		// number of placeholders. It's safe because we just taken len of event
+		// types to query, values of event types are passed as parameters.
+		eventsTypesInQuery := fmt.Sprintf(" AND event_type IN (%s)",
+			// Create following part: `?,?,?,?` based on len of eventTypes.
+			strings.TrimSuffix(strings.Repeat("?,", len(params.filter.eventTypes)), ","))
+		qb.Append(eventsTypesInQuery,
+			sliceWithTicks(params.filter.eventTypes)...,
+		)
+	}
+
+	if params.order == types.EventOrderAscending {
+		if params.startKeyset != nil {
+			qb.Append(` AND (event_time, uid) > (?,?)`,
+				fmt.Sprintf("timestamp '%s'", params.startKeyset.t.Format(athenaTimestampFormat)), fmt.Sprintf("'%s'", params.startKeyset.uid.String()))
+		}
+
+		qb.Append(` ORDER BY event_time ASC, uid ASC`)
+	} else {
+		if params.startKeyset != nil {
+			qb.Append(` AND (event_time, uid) < (?,?)`,
+				fmt.Sprintf("timestamp '%s'", params.startKeyset.t.Format(athenaTimestampFormat)), fmt.Sprintf("'%s'", params.startKeyset.uid.String()))
+		}
+		qb.Append(` ORDER BY event_time DESC, uid DESC`)
+	}
+
+	qb.Append(` LIMIT ?`, strconv.Itoa(params.limit))
+
+	return qb.String(), qb.Args()
+}
+
+func (q *querier) startQueryExecution(ctx context.Context, query string, params []string) (string, error) {
+	startQueryInput := &athena.StartQueryExecutionInput{
+		QueryExecutionContext: &athenaTypes.QueryExecutionContext{
+			Database: aws.String(q.database),
+		},
+		ExecutionParameters: params,
+		QueryString:         aws.String(query),
+	}
+	if q.workgroup != "" {
+		startQueryInput.WorkGroup = aws.String(q.workgroup)
+	}
+
+	if q.queryResultsS3 != "" {
+		startQueryInput.ResultConfiguration = &athenaTypes.ResultConfiguration{
+			OutputLocation: aws.String(q.queryResultsS3),
+		}
+	}
+
+	startQueryOut, err := q.athenaCli.StartQueryExecution(ctx, startQueryInput)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return aws.ToString(startQueryOut.QueryExecutionId), nil
+}
+
+func (q *querier) waitForSuccess(ctx context.Context, queryId string) error {
+	for {
+		// TODO(tobiaszheller): on first call it makes sense to sleep for longer period because it's typically
+		// above 1s before we can start seeing results. Come back at some point and fix it.
+		select {
```

Comment:
what do you think about max wait time? This search fn don't have ctx yet so i think it will be good to provide some resonable maximum value, does 30s seems right for you?